### PR TITLE
Add code comments to QR code generation

### PR DESCRIPTION
**Pull request by Shane32**
Improves readability, given that the code to generate a QR code is very complex, consisting of nearly 1,700 lines of raw code.

Requires PR #516 to be merged first for proper GitHub comparison.

Although most of the commented methods are private, it would be beneficial to verify the comments for accuracy.  They seem fine to me.

### CODE REVIEW

File: QRCoder/QRCodeGenerator.ModulePlacer.MaskPattern.cs [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+                    score4 = Math.Min(prevMultipleOf5, nextMultipleOf5) * 10;
+
+                    // Return the sum of all four penalties
+                    return (score1 + score2) + (score3 + score4);
+                }
+
+                /// <summary>
+                /// Matches the specified pattern in QR code evaluation rules (1:1:3:1:1 ratio).
+                /// </summary>
+                private static bool MatchesPattern1(List<BitArray> matrix, int x, int y)
```
Comment (Shane32) [inline QRCoder/QRCodeGenerator.ModulePlacer.MaskPattern.cs:193]: This abstraction/deduplication of code was the only change besides adding comments.

Comment (Shane32) [reply QRCoder/QRCodeGenerator.ModulePlacer.MaskPattern.cs]: And it didn't work so I reverted it.

File: QRCoder/QRCodeGenerator.ECCLevel.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+{
+    public partial class QRCodeGenerator
+    {
+        /// <summary>
+        /// Error correction level. These define the tolerance levels for how much of the code can be lost before the code cannot be recovered.
+        /// </summary>
+        public enum ECCLevel
+        {
+            /// <summary>
+            /// 7% may be lost before recovery is not possible
```
[the remark this answers is not in the document]
Comment (Shane32) [reply QRCoder/QRCodeGenerator.ECCLevel.cs]: These are actually unchanged from what was there, so I rewrote the comments for consistency with the other updated documentation.  The rewrite has "approximately", but if you like "up to" better I can change it.

File: QRCoder/QRCodeGenerator.EncodingMode.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            /// Byte encoding mode, primarily using the ISO-8859-1 character set. Each character is encoded into 8 bits.
+            /// When combined with ECI, it can be adapted to use other character sets.
+            /// </summary>
+            Byte = 4,
+
+            /// <summary>
+            /// Kanji encoding mode, which is used to encode characters from the Shift JIS character set, primarily for Japanese Kanji and Kana characters.
+            /// One character is encoded into 13 bits.
+            /// </summary>
+            Kanji = 8,
```
[the remark this answers is not in the document]
Comment (Shane32) [reply QRCoder/QRCodeGenerator.EncodingMode.cs:32]: I agree

File: QRCoder/QRCodeGenerator.ECCLevel.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (codebude) [reply QRCoder/QRCodeGenerator.ECCLevel.cs]: Oh, than I must somehow mixed up old/new code while writing this comment. Saw your current version (approx.) and it looks good to me.

File: QRCoder/QRCodeGenerator.EncodingMode.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Shane32) [reply QRCoder/QRCodeGenerator.EncodingMode.cs:34]: Updated

**Review by codebude — APPROVED** [review]

**Comment by codebude** [comment]
Do you plan more updates/commits? Otherwise I would be fine, merging it now.

**Comment by Shane32** [comment]
No, I'd merge it now.

**Comment by Shane32** [comment]
Actually, there's one last method I'd comment.... hold on.

**Review by codebude — APPROVED** [review]

**Comment by Shane32** [comment]
I finished btw.

**Pull request merged**